tests: don't fail on changes in the http log during e2e

justinsb opened the pull request:
Because we don't want to treat a new upstream field as an error here.

yuwenma commented on pkg/test/utils.go:
I can see that we may want to bypass the real HTTP headers, and maybe some dynamic field like timestamp(which mostly have been handled in normalize.go), I'm a little bit concerned about passing the entire http golden log. 

Mind sharing more on which http part that you want to skip?

justinsb replied:
So this is only in the e2e tests against real GCP.  We want to capture the logs, but the problem is that if there's an undocumented/new field being returned by GCP, we don't necessarily want to fail our e2e.

I do see the point though, it would be nice if we could get a failure signal here.  I think the problem is that some of the fields are not yet in proto, but I can give it a go....

yuwenma replied:
I see. That makes sense. Looks like the ONLY_WARN_ON_GOLDEN_DIFFS on-and-off can be one of the new field detector.

justinsb commented:
Rebased because of conflicting merge.

Also added a comment about why we need this - because otherwise adding a field in the upstream API breaks our tests.  Maybe we can get there one day, but it's a big reach right now.

/assign @cheftako

yuwenma commented:
/lgtm

justinsb commented:
/hold cancel
